res/controllers/Numark-Party-Mix.scripts.js

```diff
@@ -0,0 +1,440 @@
+                midi.sendShortMsg(0x93 + deckNumber, 0x17, 0x7F); // cue
+            } else {
+                midi.sendShortMsg(0x83 + deckNumber, 0x17, 0x01); // cue
+            }
+        },
+        outConnect: false
+    });
+};
+NumarkPartyMix.ModeEFX.prototype = Object.create(components.ComponentContainer.prototype);
+
+NumarkPartyMix.Browse = function() {
+    var pressturn = false;
+
+    this.knob = new components.Encoder({
+        input: function(channel, control, value) {
+            var direction;
+            if (pressturn) {
+                if (value > 0x40) {
+                    engine.setParameter("[Library]", "MoveFocusForward", 1);
+                } else {
+                    engine.setParameter("[Library]", "MoveFocusBackward", 1);
+                }
+            } else {
+                direction = (value > 0x40) ? -1 : 1;
+                engine.setParameter("[Library]", "MoveVertical", direction);
+            }
+        }
+    });
+
+    this.knobButton = new components.Button({
+        group: "[Library]",
+        type: components.Button.prototype.types.powerWindow,
+        isLongPressed: false,
+        input: function(channel, control, value, status, group) {
+            pressturn = this.isPress(channel, control, value, status);
+            print("pressturn");
+            print(pressturn);
+            if (pressturn) {
+                this.inToggle(channel, control, value, status, group);
+                this.isLongPressed = false;
+                this.longPressTimer = engine.beginTimer(this.longPressTimeout, function() {
+                    this.isLongPressed = true;
+                    this.longPressTimer = 0;
+                }, true);
+            } else {
+                this.inToggle(channel, control, value, status, group);
+                if (this.longPressTimer !== 0) {
+                    engine.stopTimer(this.longPressTimer);
+                    this.longPressTimer = 0;
+                }
+                this.isLongPressed = false;
+            }
+        },
+        inToggle: function(channel, control, value, status, group) {
+            if (! (pressturn || this.isLongPressed)) {
+                engine.setParameter("[Library]", "GoToItem", 1);
+            }
+        }
+    });
+};
+NumarkPartyMix.Browse.prototype = new components.ComponentContainer();
+
+NumarkPartyMix.Gains = function() {
+    this.mainGain = new components.Pot({
+        group: "[Master]",
+        inKey: "gain"
+    });
+
+    this.cueGain = new components.Pot({
+        group: "[Master]",
+        inKey: "headGain"
+    });
+
+    this.cueMix = new components.Pot({
+        group: "[Master]",
+        inKey: "headMix"
+    });
+};
+NumarkPartyMix.Gains.prototype = new components.ComponentContainer();
+
+NumarkPartyMix.wheelTurn = function(channel, control, value, status, group) {
+    var newValue = value;
+    if (value >= 64) {
+        // correct the value if going backwards
+        newValue -= 128;
```

Comment:
What values does the wheel send when being turned backward & forward?

Reply:
clockwise: 0x01 to 0x06
counter-clockwise: 0x7F to 0x7A

Reply:
so clockwise (slow-fast) 0x01 - 0x06
counter-clockwise (slow-fast) 0x7F - 0x7A
?

either way, please add a comment that explains the input values.

Reply:
done